### PR TITLE
Maven configuration simplification

### LINKED ISSUE #353: Maven configuration simplification
As a developer, I would like to simplify the maven configuration files in `spring-cloud-stream` and its sub-projects.

Currently, the project version needs to be specified in many places in POM files though those projects can simply use the `project.version` from the parent projects. This is difficult to manage in future.

Also, there are some duplicate properties which can be removed.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Use `project.version` wherever possible
 - Move kafka related dependencies from parent `dependencyManagement` to `kafka` binder project
 - Remove duplicate properties
 - Fix typo

This resolves #353

### CODE REVIEW

File: spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/pom.xml [hunk 1]
```diff
@@ -10,11 +10,12 @@
 	<parent>
 		<groupId>org.springframework.cloud</groupId>
 		<artifactId>spring-cloud-stream-binders-parent</artifactId>
-		<version>1.0.0.BUILD-SNAPSHOT</version>
+		<version>${project.version}</version>
 	</parent>
 
 	<properties>
 		<kafka.version>0.8.2.2</kafka.version>
```
Comment (mbogoevici) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/pom.xml]: I believe these version properties need to move to the parent, along with the <dependencyManagement> statements, leaving only <dependencies> in modules.

File: spring-cloud-stream-binders/pom.xml [hunk 2]
```diff
@@ -25,48 +22,6 @@
 		<module>spring-cloud-stream-binder-kafka</module>
 	</modules>
 	
-	<dependencyManagement>
```
Comment (mbogoevici) [inline spring-cloud-stream-binders/pom.xml]: As per my other comment, let's consolidate this in a single <depenndencyManagement> section.

**Comment by ilayaperumalg** [comment]
@sobychacko The problem with that approach is that when we install the individual POMs of each child projects, the `project.version` doesn't get set. May be there is an option to set when doing install?

**Comment by sobychacko** [comment]
LGTM - rebased, squashed and merged with 4dfd217db6e7580ae67b5889a1c119d116720926

**Pull request closed without merging**